File: python/tracing/langchain/01_chat_model_invoke.py
"""Chat model invoke."""

from langchain_core.language_models.fake_chat_models import FakeListChatModel
from langchain_core.messages import HumanMessage, SystemMessage

from _shared import flush, init_telemetry, message_text, tracing_config


def chat_model_invoke() -> None:
    telemetry = init_telemetry("langchain-chat-model-invoke")
    model = FakeListChatModel(responses=["Bonjour, Respan."])
    try:
        response = model.invoke(
            [
                SystemMessage(content="Translate English to French."),
                HumanMessage(content="Hello, Respan."),
            ],
            config=tracing_config("chat_model_invoke"),
        )
        print(message_text(response))
    finally:
        flush(telemetry)


if __name__ == "__main__":
    chat_model_invoke()

File: typescript/tracing/langchain/01_chat_model_invoke.ts
import { HumanMessage, SystemMessage } from "@langchain/core/messages";

import { fakeChat, initRespan, messageText, shutdown, tracingConfig } from "./_shared";

export async function chatModelInvoke(): Promise<void> {
  const runtime = await initRespan("typescript-langchain-chat-model-invoke");
  const model = fakeChat(["Bonjour, Respan."]);

  try {
    const response = await model.invoke(
      [
        new SystemMessage("Translate English to French."),
        new HumanMessage("Hello, Respan."),
      ],
      tracingConfig(runtime, "chat_model_invoke"),
    );
    console.log(messageText(response));
  } finally {
    await shutdown(runtime);
  }
}

if (import.meta.url === `file://${process.argv[1]}`) {
  await chatModelInvoke();
}
